Makefile

```
@@ -381,34 +389,69 @@ ifeq ($(EXPORT_RESULT), true)
 	gocov convert profile.cov | gocov-xml > coverage.xml
 endif
 
-.PHONY: test-clusterctl-create
+.PHONY: test-cluster-create
 test-clusterctl-create: ## Run the tests using clusterctl
 	which clusterctl
 	clusterctl version
 	clusterctl config repositories | grep nutanix
 	clusterctl generate cluster ${TEST_CLUSTER_NAME} -i nutanix:${LOCAL_PROVIDER_VERSION} --list-variables -v 10
 	clusterctl generate cluster ${TEST_CLUSTER_NAME} -i nutanix:${LOCAL_PROVIDER_VERSION} --target-namespace ${TEST_NAMESPACE}  -v 10 > ./cluster.yaml
-	kubectl create ns $(TEST_NAMESPACE) || true
-	kubectl apply -f ./cluster.yaml -n $(TEST_NAMESPACE)
+	kubectl create ns $(TEST_NAMESPACE) --dry-run=client -oyaml | kubectl apply --server-side -f -
+	kubectl apply --server-side -f ./cluster.yaml
 
-.PHONY: test-clusterctl-delete
+.PHONY: test-cluster-delete
 test-clusterctl-delete: ## Delete clusterctl created cluster
-	kubectl -n ${TEST_NAMESPACE} delete cluster ${TEST_CLUSTER_NAME}
+	kubectl -n ${TEST_NAMESPACE} delete cluster ${TEST_CLUSTER_NAME} --ignore-not-found
 
-.PHONY: test-kubectl-bootstrap
-test-kubectl-bootstrap: ## Run kubectl queries to get all capx management/bootstrap related objects
+.PHONY: list-bootstrap-resources
+list-bootstrap-resources: ## Run kubectl queries to get all capx management/bootstrap related objects
 	kubectl get ns
 	kubectl get all --all-namespaces
 	kubectl -n capx-system get all
 	kubectl -n $(TEST_NAMESPACE) get Cluster,NutanixCluster,Machine,NutanixMachine,KubeAdmControlPlane,MachineHealthCheck,nodes
 	kubectl -n capx-system get pod
 
-.PHONY: test-kubectl-workload
-test-kubectl-workload: ## Run kubectl queries to get all capx workload related objects
+.PHONY: list-workload-resources
+list-workload-resources: ## Run kubectl queries to get all capx workload related objects
 	kubectl -n $(TEST_NAMESPACE) get secret
 	kubectl -n ${TEST_NAMESPACE} get secret ${TEST_CLUSTER_NAME}-kubeconfig -o json | jq -r .data.value | base64 --decode > ${TEST_CLUSTER_NAME}.workload.kubeconfig
 	kubectl --kubeconfig ./${TEST_CLUSTER_NAME}.workload.kubeconfig get nodes,ns
 
+.PHONY: test-cc-cluster-create
+test-cc-cluster-create: cluster-templates
+	clusterctl generate cluster ${TEST_CLUSTER_CLASS_NAME} --from ./templates/cluster-template-clusterclass.yaml -n $(TEST_NAMESPACE) > ${TEST_CLUSTER_CLASS_NAME}.yaml
+	clusterctl generate cluster ${TEST_TOPOLOGY_CLUSTER_NAME} --from ./templates/cluster-template-topology.yaml -n $(TEST_NAMESPACE) > ${TEST_TOPOLOGY_CLUSTER_NAME}.yaml
```

>Error: value for variables [CLUSTER_CLASS_NAME] is not set. Please set the value using os environment variables or the clusterctl config file
make: *** [Makefile:422: test-cc-cluster-create] Error 1

Ran into the above error, something like this will work.

```suggestion
	CLUSTER_CLASS_NAME=${TEST_CLUSTER_CLASS_NAME} clusterctl generate cluster ${TEST_CLUSTER_CLASS_NAME} --from ./templates/cluster-template-clusterclass.yaml -n $(TEST_NAMESPACE) > ${TEST_CLUSTER_CLASS_NAME}.yaml
	CLUSTER_CLASS_NAME=${TEST_CLUSTER_CLASS_NAME} clusterctl generate cluster ${TEST_TOPOLOGY_CLUSTER_NAME} --from ./templates/cluster-template-topology.yaml -n $(TEST_NAMESPACE) > ${TEST_TOPOLOGY_CLUSTER_NAME}.yaml
```

---

we should put it in clusterctl.yaml.tmpl

---

👍 yep lets put in in the `clusterctl.yaml.tmpl` along with the other vars.